### ISSUE TITLE
Unified pipeline for data conversion

### ISSUE THREAD

**Issue by albertmeronyo** [CLARIAH/wp4-requirements#1]

**Comment by albertmeronyo** [comment]
Even if only covers a few cases completely.

**Comment by albertmeronyo** [comment]
Stash older versions of datasets on publication of new versions

**Comment by RinkeHoekstra** [comment]
More detailed description please!

**Comment by albertmeronyo** [comment]
Currently, the data conversion pipeline is too fragmented and requires human intervention at several stages. The current workflow is:
- User finds an inefficient way (e.g. by sending it to us) of uploading his/her CSVs to the Hub
- These become available in QBer, so coding can start (this part is hence nicely automatized)
- Once the user is done with coding, it's unclear how the pipeline works between clicking 'publish' on QBer, and getting harmonized datasets on the triplestore ready to query. For this latter step, the generated JSON mappings must be combined with the result of the converters on the CSV files. Then, this 'harmonized' dataset must be loaded into the triplestore. Intermediate steps on this process may require materializing some or all of the data (raw, RDF-converted, mappings, etc.) into the Gitlab instance.

**Comment by RinkeHoekstra** [comment]
So this means that:
- Users can upload a CSV via QBer
  - There should be an upload UI for QBer, this is CLARIAH/qber#61
  - QBer should post the CSV to the API
- This CSV will be deposited in GitLab, the API will return a JSON structure for the dataset to Qber, including the commit hash
- QBer workflow then continues as normal
- The user submits the annotated file via the normal way
- The API then converts the CSV to RDF based on the mappings in the JSON
- The RDF is also deposited in GitLab
- The RDF is posted to the triple store
- The API returns a success message + a link to the RDF data

**Comment by RinkeHoekstra** [comment]
![img_2485](https://cloud.githubusercontent.com/assets/715019/14636878/5262c3de-062e-11e6-9639-1557fc836c99.jpg)

**Comment by RinkeHoekstra** [comment]
Milestone meeting:

Turns out this is really close to #24 
- Weird to hack it now, and redo it later
- But... the data team currently has no means to transpose data into RDF (other than what they had before): there is no **approved** way.

Resolution is to close this issue and make a separate issue for the CSVW conversion pipeline completion (i.e. trigger conversion & upload based on GitLab commits of mapping file).

This new issue is #37
